cmd/endpointQuerier/main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"context"
+	"encoding/xml"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptrace"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"../../internal/endpoints"
+	"../../internal/fhir"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Metrics collected inside build-in prometheus vector. Each METRIC has its own registrations
+var httpCodesGaugeVec *prometheus.GaugeVec
+var responseTimeGaugeVec *prometheus.GaugeVec
+var tlsVersionGaugeVec *prometheus.GaugeVec
+var fhirVersionGaugeVec *prometheus.GaugeVec
+var totalUptimeChecksCounterVec *prometheus.CounterVec
+var totalFailedUptimeChecksCounterVec *prometheus.CounterVec
+
+// Need to define timeout or else it is infinite
+var netClient = &http.Client{
+	Timeout: time.Second * 35,
+}
+
+func getHTTPRequestTimingFor(url string, organizationName string, recordLongRunning bool) {
+	// Specifically query the FHIR endpoint metadata
+	url += "metadata"
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		println("HTTP Request Error: %s", err)
+	}
+
+	var start time.Time
+	trace := &httptrace.ClientTrace{}
+
+	// Drop connection if no reply within 30 seconds
+	ctx, cancel := context.WithDeadline(req.Context(), time.Now().Add(30*time.Second))
+	// Cancel the context once we are done with this function so that context does not remain in memory (causing a leak)
+	defer cancel()
+	req = req.WithContext(httptrace.WithClientTrace(ctx, trace))
+
+	start = time.Now()
+	resp, err := netClient.Do(req)
+	responseTimeGaugeVec.WithLabelValues(endpoints.NamespaceifyString(organizationName)).Set(float64(time.Since(start).Seconds()))
+
+	// Need to think about wether or not an errored request is considered a failed uptime check
+	if err != nil || (resp != nil && resp.StatusCode != http.StatusOK) {
```

Comment:
man - this is confusing to read. I see that go requires booleans in its if statement (makes sense), which means you can't just say `if err || (resp && resp.StatusCode != http.StatusOK)` :/ maybe there's another way this can be rewritten. No troubles if not.

Reply:
To me I read this as "if there is an error or there is a non 200 status code" `resp != nil` needs to be in there so that we don't try to check nil.StatusCode